editor/src/components/context-menu-wrapper.tsx

```diff
@@ -18,206 +27,195 @@ import type { WindowPoint } from '../core/shared/math-utils'
+  }, [items])
 
-  isDisabled = (item: ContextMenuItem<T>): (() => boolean) => {
-    return () => {
-      if (typeof item.enabled === 'function') {
-        return !item.enabled(this.props.getData())
-      } else {
-        return !item.enabled
+  const isHidden = useCallback(
+    (item: Item<T>): (() => boolean) => {
+      return () => {
+        if (typeof item?.isHidden === 'function') {
+          return item.isHidden(getData())
+        }
+        return item?.isHidden ?? false
       }
-    }
-  }
+    },
+    [getData],
+  )
 
-  renderItem(item: ContextMenuItem<T>, index: number) {
-    return (
-      <Item
-        key={`context-menu-${index}-item`}
-        disabled={this.isDisabled(item)}
-        // eslint-disable-next-line react/jsx-no-bind
-        onClick={({ event, triggerEvent }) => {
-          event.stopPropagation()
-          const rightClickCoordinate: WindowPoint | null = (() => {
-            if (!(triggerEvent instanceof MouseEvent)) {
-              return null
-            }
-            return windowPoint({ x: triggerEvent.clientX, y: triggerEvent.clientY })
-          })()
-          item.action(this.props.getData(), this.props.dispatch, rightClickCoordinate, event)
-          contextMenu.hideAll()
-        }}
-        hidden={this.isHidden(item)}
-        style={{
-          height: item.isSeparator ? 9 : 28,
-          display: 'flex',
-          alignItems: 'center',
-          borderRadius: 4,
-        }}
-      >
-        <span style={{ flexGrow: 1, flexShrink: 0 }} className='react-contexify-span'>
-          {item.name}
-        </span>
-        <span style={{ flexGrow: 0, flexShrink: 0, opacity: 0.6 }} className='shortcut'>
-          {item.shortcut}
-        </span>
-      </Item>
-    )
-  }
+  const isDisabled = useCallback(
+    (item: Item<T>): (() => boolean) =>
+      () => {
+        return typeof item?.enabled === 'function' ? !item.enabled(getData?.()) : !item?.enabled
+      },
+    [getData],
+  )
 
-  onShown = () => {
-    document.body.classList.add(BodyMenuOpenClass)
-  }
-  onHidden = () => {
-    document.body.classList.remove(BodyMenuOpenClass)
-  }
+  const renderItem = useCallback(
+    (item: Item<T>, index: number) => {
+      return (
+        <Item
+          key={`context-menu-${index}-item`}
+          disabled={isDisabled(item)}
+          // eslint-disable-next-line react/jsx-no-bind
+          onClick={({ event, triggerEvent }) => {
+            event.stopPropagation()
+            const rightClickCoordinate: WindowPoint | null = (() => {
+              if (!(triggerEvent instanceof MouseEvent)) return null
+              return windowPoint({ x: triggerEvent.clientX, y: triggerEvent.clientY })
+            })()
+            item?.action(getData(), dispatch, rightClickCoordinate, event)
+            contextMenu.hideAll()
+          }}
+          hidden={isHidden(item)}
+          style={{
+            height: item?.isSeparator ? 9 : 28,
+            display: 'flex',
```

Comment:
Maybe too soon / out of scope for this refactor: but I wonder if we'd be better off with a `grid` here, or even a variable for the list-item layout that we want (just label, label with accesor, icon+ label, icon+label+accessor etc).